miio/protocol.py

```diff
@@ -161,6 +161,11 @@ def _encode(self, obj, context, path):
 
     def _decode(self, obj, context, path) -> Union[Dict, bytes]:
         """Decrypts the payload using the token stored in the context."""
+        # if there is no payload, decode to 0 bytes. Missing payload is expected for discovery messages.
+        # note that we don't have "token" in the context for discovery replies so we couldn't decode it
+        # anyway.
```

Comment:
```suggestion
        # Missing payload is expected for discovery messages.
```
let's keep it simple here, also, there are some (very) old devices that actually respond with a valid token :-)